core/src/main/java/org/apache/gravitino/storage/relational/mapper/UserMetaBaseSQLProvider.java

```diff
@@ -138,6 +138,21 @@ public String listUsersByRoleId(@Param("roleId") Long roleId) {
         + " AND us.deleted_at = 0 AND re.deleted_at = 0";
   }
 
+  public String listUserPOsByMetalake(@Param("metalakeName") String metalakeName) {
+    return "SELECT ut.user_id as userId, ut.user_name as userName,"
+        + " ut.metalake_id as metalakeId,"
+        + " ut.audit_info as auditInfo,"
+        + " ut.current_version as currentVersion, ut.last_version as lastVersion,"
+        + " ut.deleted_at as deletedAt"
+        + " FROM "
+        + USER_TABLE_NAME
+        + " ut JOIN "
+        + MetalakeMetaMapper.TABLE_NAME
+        + " mt ON ut.metalake_id = mt.metalake_id"
+        + " WHERE mt.metalake_name = #{metalakeName}"
+        + " AND ut.deleted_at = 0 AND mt.deleted_at = 0";
+  }
```

Comment:
This SQL will not throw an exception when metalake is not existed, it will only return an empty list. In your definition, you have `NoSuchMetalakeException` defined, to throw an such exception, you have to add one more check to see if metalake is existed or not.

Reply:
Added the `checkMetalakeExists`.